tests/stdlib/tarithmetics.nim

```diff
@@ -0,0 +1,44 @@
+discard """
+  targets: "c cpp js"
+"""
+
+# TODO: in future work move existing arithmetic tests here, where they belong: Nim/tests/arithm
+# FYI https://github.com/nim-lang/Nim/pull/17767
+
+template main =
+  # put all arithmetic tests
+
+  block tshr:
+
+    # Signed types
+    block:
+      let
+        t0: int = -3 shr 2
+        t1: int8 = -4'i8 shr 1'i8
+        t2: int16 = -5'i16 shr 1'i16
+        t3: int32 = -7'i32 shr 1'i32
```

Comment:
the compiler does const folding for `-7'i32 shr 1'i32`, so that code isn't being run at RT.
so maybe something like:
```
let a1 = -4'i8
let a2 = 1'i8
doAssert a1 shr a2 == -d'i8
```
etc

Reply:
sorry, I have not comprehended the conversation just yet. 
where is the specification you advised below ?

And I can't figure out that the sample code you pointed out for int32 is int8. Also, is the d in the code correct for a hexadecimal d?

```nim
let a1 = -4'i8
let a2 = 1'i8
doAssert a1 shr a2 == -d'i8 # doAssert -2 == -100 ?
```

Reply:
use `a1 shr a2 == -2'i8` I think

Reply:
yes, sorry for the typo, i corrected it above. 

for more details, see `ctfeWhitelist` in compiler/ast.nim: it includes `mShrI, mShlI` etc, which means compiler will evaluate `a shl b` at CT if a, b are known at CT, which means it won't test RT code (eg the one generated by jsgen)

Reply:
Thank you. I'll take your advice and fix it.

Reply:
you can verify whether it worked by inspecting the generated js code and making sure it contains `>>>` for those statements; right now (as i predicted) it doesn't, due to const folding

Reply:
```nim
  when true:
    import std/jsconsole
    let
      a = -3
      b = -2
      t0: int = a shr b
    console.log(t0)
```

above code generated following code.

```js
/* Generated by the Nim Compiler v1.5.1 */
var framePtr = null;
var excHandler = 0;
var lastJSError = null;
if (!Math.trunc) {
  Math.trunc = function(v) {
    v = +v;
    if (!isFinite(v)) return v;
    return (v - v % 1) || (v < 0 ? -0 : v === 0 ? v : 0);
  };
}

var a_369098753 = -3;
var b_369098754 = -2;
var t0_369098759 = (a_369098753 >> b_369098754);
console.log(t0_369098759);
```

but const declation generate following code
```nim
  when true:
    import std/jsconsole
    let
      t0: int = -3 shr -2
    console.log(t0
```

```js
/* Generated by the Nim Compiler v1.5.1 */
var framePtr = null;
var excHandler = 0;
var lastJSError = null;
if (!Math.trunc) {
  Math.trunc = function(v) {
    v = +v;
    if (!isFinite(v)) return v;
    return (v - v % 1) || (v < 0 ? -0 : v === 0 ? v : 0);
  };
}

var t0_369098757 = -1;
console.log(t0_369098757);
```
I've comprehended the system.
I've comprehended the system. Thank you so much ！